client/express-checkout/utils/index.ts

```diff
@@ -0,0 +1,104 @@
+export * from './normalize';
+
+/**
+ * An /incomplete/ representation of the data that is loaded into the frontend for the Express Checkout.
+ */
+interface WCPayExpressCheckoutParams {
+	ajax_url: string;
+
+	/**
+	 * Express Checkout Button style configuration.
+	 */
+	button: {
+		type: string;
+		theme: string;
+		height: string;
+		locale: string;
+		branded_type: string;
+	};
+
+	/**
+	 * Indicates in which context the button is being displayed.
+	 */
+	button_context: 'checkout' | 'cart' | 'product' | 'pay_for_order';
+	checkout: {
+		country_code: string;
+		currency_code: string;
+		needs_payer_phone: boolean;
+		needs_shipping: boolean;
+	};
+
+	/**
+	 * Indicaters whether the page has a Cart or Checkout Block on it.
+	 */
+	has_block: boolean;
+
+	/**
+	 * True if we're on the checkout page.
+	 */
+	is_checkout_page: boolean;
+
+	/**
+	 * True if we're on a product page.
+	 */
+	is_product_page: boolean;
+
+	/**
+	 * True if we're on the pay for order page.
+	 */
+	is_pay_for_order_page: boolean;
+	nonce: {
+		add_to_cart: string;
+		checkout: string;
+		empty_cart: string;
+		get_cart_details: string;
+		get_selected_product_data: string;
+		pay_for_order: string;
+		platform_tracker: string;
+		shipping: string;
+		update_shipping: string;
+	};
+
+	/**
+	 * Product specific options.
+	 */
+	product: {
+		needs_shipping: boolean;
+		currency: string;
+	};
+	stripe: {
+		accountId: string;
+		locale: string;
+		publishableKey: string;
+	};
+	total_label: string;
+	wc_ajax_url: string;
+}
+
+declare global {
+	interface Window {
+		wcpayExpressCheckoutParams?: WCPayExpressCheckoutParams;
+	}
+}
+
+export const getExpressCheckoutData = (
+	key: keyof WCPayExpressCheckoutParams
+) => {
+	if ( window.wcpayExpressCheckoutParams ) {
+		return window.wcpayExpressCheckoutParams?.[ key ];
+	}
+
+	return null;
+};
+
+/**
+ * Get error messages from WooCommerce notice from server response.
+ *
+ * @param notice Error notice.
+ * @return Error messages.
+ */
+export const getErrorMessageFromNotice = ( notice: string ) => {
```

Comment:
It seems like we're not using this, but the function defined in `client/utils/express-checkout/index.js`.

Also, should we fix these linter warnings?

Reply:
Ah, I imported the wrong function, fixed in 54eb67b25eb1f17fb6167cb1498be7bdc794f481.

And no, the linter is wrong; we [explicitly encourage using TypeScript's type inference](https://github.com/Automattic/woocommerce-payments/blob/develop/docs/typescript/general-conventions.md#rely-on-type-inference) so I turned off the rule in eb3ecd3036ed2503cdb54bcdab3fc470d37aa918 instead.